AmplifyPlugins/Predictions/AWSPredictionsPlugin/Support/Internal/NativeWebSocketProvider.swift

```diff
@@ -72,10 +72,15 @@ class NativeWebSocketProvider: NSObject, AWSTranscribeStreamingWebSocketProvider
             switch result {
             case .failure(let error):
                 let status = AWSTranscribeStreamingClientConnectionStatus.closed
+                let nsError = error as NSError?
 
                 self.callbackQueue.async {
                     self.clientDelegate.connectionStatusDidChange(status, withError: error)
                 }
+
+                guard nsError?.domain != NSURLErrorDomain else {
```

Comment:
If the error is of type NSURLErrorDomain, we do not want to keep calling `listen()`, we should just return

Reply:
Let's not use -1003 as a literal, though. Use the appropriate error code